wrappers/jira_client.py

```diff
@@ -25,23 +28,31 @@ def convert_seconds_to_jira_time(seconds):
 
         return " ".join(jira_time) if jira_time else "0s"
 
-    def __init__(self, config):
+    def __init__(self, config: dict):
         self.jira_url = config.get('jira_url')
-        self.tempo_api = config.get('tempo_api')
-        self.tempo_user = config.get('tempo_user')
-        self.tempo_password = config.get('tempo_password')
-        self.session = self.init_session()
-
-    def init_session(self):
+        self.tempo_url = config.get('tempo_url')
+        self.worklog_url = urljoin(self.tempo_url, 'worklogs')
+        self.jira_api_token = config.get('jira_api_token')
+        self.tempo_api_token = config.get('tempo_api_token')
+        self.jira_account_email = config.get('jira_account_email')
+        self.jira_session, self.account_id = self.init_jira_session()
+        self.tempo_session = self.init_tempo_session()
+
+    def init_jira_session(self):
         session = requests.Session()
-        session.auth = (self.tempo_user, self.tempo_password)
+        session.headers.update({
+            "Accept": "application/json",
+        })
+        session.auth = HTTPBasicAuth(
+            self.jira_account_email,
+            self.jira_api_token
+        )
 
-        api = self.jira_url + 'rest/api/2'
-        resp = session.get('%s/myself' % api)
-        if resp.ok:
-            return session
+        resp = session.get(urljoin(self.jira_url, 'rest/api/3/myself'))
+        if resp.status_code == 200:
+            return session, resp.json()["accountId"]
         elif resp.status_code == 401:
-            raise Exception("Error: Incorrect password or username.")
+            raise Exception("Error: Jira authentication failed.")
```

Comment:
I've configured everything but I reach this point and the the content of the response is `Client must be authenticated to access this resource.` .... ma dai? :smile: